accepted/00000-salt-ssh-with-salt-bundle.md

```diff
@@ -0,0 +1,95 @@
+- Feature Name: Salt-SSH with Salt Bundle
+- Start Date: 2021-12-08
+
+# Summary
+[summary]: #summary
+
+Use Salt Bundle to handle Salt-SSH sessions on the minion side.
+
+Salt Bundle is a single package containing salt codebase, Python and python modules required for salt.
+
+# Motivation
+[motivation]: #motivation
+
+Due to the deprecation of Python 2 and dropping support of it, while some of supported Linux systems still has no proper Python 3 version supported by Salt, we need to have a way to process Salt-SSH events on the minion side.
+
+The following systems are using salt codebase older than `3002.2`:
+- SLE 11 (LTSS ends 31 Mar 2022, no plans to support it with Salt Bundle)
+- SLE 12
+- RHEL 7 (and clones)
+- Debian 9
+
+We can deploy Salt Bundle with bundled Python inside the virtual environment and handle all the Salt-SSH events using Salt Bundle with making Salt-SSH independent from Python on the minion.
+Additionally we are extending the list of working modules as Salt Bundle contains binary python modules required by some of Salt modules.
+
+This approach will help us to limit the number of supported versions of Salt codebase and drop `py26-compat-salt`, `py27-compat-salt` (as these packages are shipping `2016.11.10` and `3000.3` respectively) and all Python 2 dependencies maintenance.
+So we don't need to maintain old salt codebases, but focus only on the latest one.
+
+# Detailed design
+[design]: #detailed-design
+
+## Uyuni roster
+
+We can use Uyuni roster module to provide roster data instead of creating flat roster files with Java. PR moving the logic to Uyuni roster on Java side: https://github.com/uyuni-project/uyuni/pull/4457
+
+This change can help to avoid creating temporary roster files (https://docs.saltproject.io/en/latest/topics/ssh/roster.html) and implements caching additionally.
+
+## Pre flight script
+
+Salt can execute a shell script on the salt-ssh client before the "real salt-ssh execution" starts.
+This script downloads the salt-bundle package from the appropriate bootstrap repository and extracts the virtual environment.
+By-default there is no way to pass additional parameters to pre fligt script running on the minion.
+In this case we have to generate individual pre flight script to each of the minion. The main drawback here is maintaining these pre flight script files.
+
+The other solution is to pass additional parameter to the static pre flight script.
```

Comment:
As this is an RFC, please select one of the solutions and present it as the chosen one, listing the other in alternatives with its pros/cons as a justification of the one you picked.